test-suite/testkit/tests/counter/main.rs

```diff
@@ -918,3 +921,187 @@ fn test_explorer_add_transaction_with_invalid_transaction() {
         ApiError::BadRequest(ref body) if body == error_body
     );
 }
+
+#[test]
+fn test_explorer_api_with_before_transactions_error() {
+    let (mut testkit, api) = init_testkit();
+    let (pubkey, key) = crypto::gen_keypair();
+    let tx = Increment::new(13).sign(SERVICE_ID, pubkey, &key);
+
+    // This tx lead to error in before_transaction on the next transaction
+    testkit.create_block_with_transaction(tx.clone());
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&CallStatusQuery {
+            height: Height(1),
+            id: SERVICE_ID,
+        })
+        .get::<CallStatusResponse>("v1/call_status/before_transactions")
+        .expect("Explorer Api unexpectedly failed");
+    assert!(response.status.0.is_ok());
+
+    let tx = Increment::new(1).sign(SERVICE_ID, pubkey, &key);
+    // So perform one more tx to check the error
+    testkit.create_block_with_transaction(tx.clone());
+
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&CallStatusQuery {
+            height: Height(2),
+            id: SERVICE_ID,
+        })
+        .get::<CallStatusResponse>("v1/call_status/before_transactions")
+        .expect("Explorer Api unexpectedly failed");
+    assert!(response.status.0.is_err());
+    assert!(response
+        .status
+        .0
+        .unwrap_err()
+        .description()
+        .contains("Number 13"));
+
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&TransactionQuery {
+            hash: tx.object_hash(),
+        })
+        .get::<CallStatusResponse>("v1/call_status/transaction")
+        .expect("Explorer Api unexpectedly failed 3");
```

Comment:
Nit: What's with the numbers? I think using backtrace is much more reliable to determine the error cause in tests rather than relying on messages.